src/main/java/model/FruitTransaction.java

```diff
@@ -0,0 +1,55 @@
+package model;
+
+public class FruitTransaction {
+    private Operation operation;
+    private String fruit;
+    private int quantity;
+
+    public Operation getOperation() {
+        return operation;
+    }
+
+    public void setOperation(String op) {
+        this.operation = switch (op) {
+            case "b" -> Operation.BALANCE;
+            case "s" -> Operation.SUPPLY;
+            case "p" -> Operation.PURCHASE;
+            case "r" -> Operation.RETURN;
+            default -> throw new IllegalArgumentException(
+                    "There is no Enum element, corresponding to the argument " + op);
+        };
+    }
```

Comment:
The method `setOperation(String op)` is doing more than just setting the value of `operation`. It's also converting the `String` input into an `Operation` enum. This violates the Single Responsibility Principle (SRP). It would be better to move the conversion logic to a separate method.